ag-docs-wafv2/azuredeploy.json

```diff
@@ -1,291 +1,401 @@
+            }
+        },
+        "adminPassword": {
+            "type": "SecureString",
+            "metadata": {
+                "description": "Password for the admin account on the backend servers"
             }
-          ]
-      }
-  },
-  {
-      "apiVersion": "2019-04-01",
-      "name": "[parameters('applicationGatewayName')]",
-      "type": "Microsoft.Network/applicationGateways",
-      "location": "[parameters('location')]",
-      "dependsOn": [
-        "[variables('virtualNetworkName')]",
-        "[variables('publicIPAddressName')]"
-      ],
-      "properties": {
-          "sku": {
-              "name": "[parameters('applicationGatewaySize')]",
-              "tier": "[parameters('applicationGatewayTier')]",
-              "capacity": "[parameters('capacity')]"
-          },
-          "gatewayIPConfigurations": [
-            {
-                "name": "appGatewayIpConfig",
-                "properties": {
-                    "subnet": {
-                        "id": "[variables('subnetRef')]"
+        },
+        "location": {
+            "defaultValue": "[resourceGroup().location]",
+            "type": "String",
+            "metadata": {
+                "description": "Location for all resources."
+            }
+        },
+        "vmSize": {
+            "defaultValue": "Standard_B2ms",
+            "type": "String",
+            "metadata": {
+                "description": "Size of the virtual machine."
+            }
+        }
+    },
+    "variables": {
+        "virtualMachines_myVM_name": "myVM",
+        "virtualNetworks_myVNet_name": "myVNet",
+        "net_interface": "net-int",
+        "ipconfig_name": "ipconfig",
+        "publicIPAddress": "public_ip",
+        "nsg_name": "vm-nsg",
+        "applicationGateways_myAppGateway_name": "myAppGateway",
+        "vnet_prefix": "10.0.0.0/16",
+        "ag_subnet_prefix": "10.0.0.0/24",
+        "backend_subnet_prefix": "10.0.1.0/24",
+        "AGWafPol01": "WafPol01"
+    },
+    "resources": [
+        {
+            "type": "Microsoft.Network/networkSecurityGroups",
+            "apiVersion": "2019-11-01",
+            "name": "[concat(variables('nsg_name'), copyIndex(1))]",
+            "location": "[parameters('location')]",
+            "properties": {
+                "securityRules": [
+                    {
+                        "name": "RDP",
+                        "properties": {
+                            "protocol": "TCP",
+                            "sourcePortRange": "*",
+                            "destinationPortRange": "3389",
+                            "sourceAddressPrefix": "*",
+                            "destinationAddressPrefix": "*",
+                            "access": "Allow",
+                            "priority": 300,
+                            "direction": "Inbound"
+                        }
                     }
-                }
+                ]
+            },
+            "copy": {
```

Comment:
move the copy property above the properties property in all the resources

Reply:
done